### PR TITLE
feat: (pool): Add max stake limit ends block countdown

### PR DESCRIPTION
**Pull request by memoyil**
Screenshot 2022-02-10 at 10 22 00
Screenshot 2022-02-10 at 10 22 55

### CODE REVIEW

File: src/views/Pools/components/PoolCard/CardFooter/ExpandedFooter.tsx [hunk 1]
```diff
@@ -131,9 +132,30 @@ const ExpandedFooter: React.FC<ExpandedFooterProps> = ({ pool, account }) => {
...
-          <Text small>{t('Max. stake per user')}:</Text>
-          <Text small>{`${getFullDisplayBalance(stakingLimit, stakingToken.decimals, 0)} ${stakingToken.symbol}`}</Text>
+        <Flex flexDirection="column">
+          <Flex mb="2px" justifyContent="space-between">
+            <Text small>{t('Max. stake per user')}:</Text>
+            <Text small>{`${getFullDisplayBalance(stakingLimit, stakingToken.decimals, 0)} ${
+              stakingToken.symbol
+            }`}</Text>
+          </Flex>
+          {hasPoolStarted && (
```
Comment (chefilip) [inline src/views/Pools/components/PoolCard/CardFooter/ExpandedFooter.tsx:142]: if `numberBlocksForUserLimit` is equal to Zero, should we show it ?

File: src/state/pools/helpers.ts [hunk 2]
```diff
@@ -43,6 +52,7 @@ export const transformPool = (pool: SerializedPool): DeserializedPool => {
     userData: transformUserData(userData),
     totalStaked: new BigNumber(totalStaked),
     stakingLimit: new BigNumber(stakingLimit),
+    numberBlocksForUserLimit: new BigNumber(numberBlocksForUserLimit),
```
Comment (chefilip) [inline src/state/pools/helpers.ts:55]: could we centralize `Max. stake limit ends`  calculation in here ?

**Comment by chefilip** [comment]
In general, LGTM. If we could avoid duplication, it would be perfect.

**Comment by memoyil** [comment]
> In general, LGTM. If we could avoid duplication, it would be perfect.

Done

File: src/state/pools/helpers.ts (hunk 2 above)
Comment (memoyil) [reply src/state/pools/helpers.ts]: Done

File: src/state/pools/fetchPools.ts [hunk 3]
```diff
@@ -41,11 +42,11 @@ export const fetchPoolsBlockLimits = async () => {
   }, [])
 
   return poolsWithEnd.map((cakePoolConfig, index) => {
-    const [startBlock, endBlock] = startEndBlockResult[index]
+    const [[startBlock], [endBlock]] = startEndBlockResult[index]
     return {
       sousId: cakePoolConfig.sousId,
-      startBlock: new BigNumber(startBlock).toJSON(),
-      endBlock: new BigNumber(endBlock).toJSON(),
+      startBlock: startBlock.toNumber(),
```
Comment (memoyil) [inline src/state/pools/fetchPools.ts:48]: It should be number according to CorePoolProps interface

**Review by chefilip — APPROVED** [review]

**Comment by chefilip** [comment]
@memoyil Could you help me resolve conflict so I could merge this awesome UI feature ? Thanks for your contribution.

**Comment by memoyil** [comment]
> @memoyil Could you help me resolve conflict so I could merge this awesome UI feature ? Thanks for your contribution.

Done

**Review by chef-jojo — APPROVED** [review]

**Review by chef-jojo — DISMISSED** [review]

**Comment by memoyil** [comment]
@chef-jojo I solved the conflicts

**Review by chefilip — APPROVED** [review]
LGTM

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**